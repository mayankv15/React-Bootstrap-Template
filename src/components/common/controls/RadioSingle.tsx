import React from "react";
import {
  Radio as MuiCheckbox,
  FormControlLabel,
  FormGroup,
} from "@mui/material";
import {
  CheckBox,
  CheckBoxOutlineBlank,
  CheckCircle,
  RadioButtonUnchecked,
} from "@mui/icons-material";
import Label from "./Label";

const RadioSingle = (props) => {
  const { label, color, onClick, view, ...other } = props;
  return (
    <FormGroup>
      <FormControlLabel
        control={
          <MuiCheckbox
            color={color ? color : "default"}
            checkedIcon={view === "circle" ? <CheckCircle /> : <CheckBox />}
            icon={
              view === "circle" ? (
                <RadioButtonUnchecked />
              ) : (
                <CheckBoxOutlineBlank />
              )
            }
          />
        }
        label={<Label title={label} />}
        {...other}
      />
    </FormGroup>
  );
};

export default RadioSingle;
