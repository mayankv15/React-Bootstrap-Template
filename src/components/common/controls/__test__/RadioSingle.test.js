import React from "react";
import { render } from "@testing-library/react";
import RadioSingle from "../RadioSingle"

describe("Renders RadioSingle", () => {
    it("should render RadioSingle correctly", () => {
        const component = render(<RadioSingle label="Above 18" view="circle" checked />);
        expect(component).toMatchSnapshot();
    });
});
