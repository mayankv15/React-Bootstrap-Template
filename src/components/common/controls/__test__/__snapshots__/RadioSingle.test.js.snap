// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders RadioSingle should render RadioSingle correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiFormGroup-root"
      >
        <label
          class="MuiFormControlLabel-root"
        >
          <span
            aria-disabled="false"
            class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root PrivateSwitchBase-checked-2 Mui-checked"
          >
            <span
              class="MuiIconButton-label"
            >
              <input
                checked=""
                class="PrivateSwitchBase-input-4"
                type="radio"
                value=""
              />
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
          >
            <p
              class="MuiTypography-root makeStyles-label-5 undefined MuiTypography-body1"
            >
              Above 18
               
              
            </p>
          </span>
        </label>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiFormGroup-root"
    >
      <label
        class="MuiFormControlLabel-root"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-1 MuiRadio-root PrivateSwitchBase-checked-2 Mui-checked"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input-4"
              type="radio"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root makeStyles-label-5 undefined MuiTypography-body1"
          >
            Above 18
             
            
          </p>
        </span>
      </label>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
